pallet-chainlink-feed/src/lib.rs

```diff
@@ -737,16 +747,29 @@ pub mod pallet {
 		/// - Removes the details for the previous round if it was superseded.
 		///
 		/// Limited to the oracles of a feed.
-		#[pallet::weight(T::WeightInfo::submit_opening_round_answers().max(
-		T::WeightInfo::submit_closing_answer(T::OracleCountLimit::get())
-		))]
+		#[pallet::weight(
+            <SubmitWeight<T>>::new(
+                T::WeightInfo::submit_opening_round_answers().max(
+		            T::WeightInfo::submit_closing_answer(T::OracleCountLimit::get())
+                ),
+                oracle.clone(),
+                *feed_id,
+                *round_id,
+            ))]
 		pub fn submit(
 			origin: OriginFor<T>,
+			// Specified oracle account
+			//
+			// TODO:
+			//
+			// This field is for calculating weights currently, remove this
+			// field after https://github.com/paritytech/substrate/issues/6316
+			oracle: T::AccountId,
```

Comment:
Oh I just noticed this, this is cumbersome and would allow submitters malicously spam other accounts ids here.

There is a correct way to do this:
https://substrate.dev/docs/en/knowledgebase/runtime/fees#post-dispatch-weight-correction

https://substrate.dev/docs/en/knowledgebase/learn-substrate/weight#post-dispatch-weight-correction

So we need basically move the weight calculation form the annotation inside this extrinsic

Reply:
awesome!